torchaudio/csrc/sox/utils.h

```diff
@@ -34,6 +34,19 @@ std::vector<std::string> list_write_formats();
 // Utilities for sox_io / sox_effects implementations
 ////////////////////////////////////////////////////////////////////////////////
 
+const std::string ENCODING_UNKNOWN = "UNKNOWN";
```

Comment:
I'd turn this into an enum in a separate header and add conversion functions to/from strings as well.

Reply:
That'll also allow you to use switch statements which probably cleans things up a bit and you're also not relying on string comparisons to do control flow.